### PR TITLE
Bridging OSS and EE serialization compatibility

### PR DESCRIPTION
**Pull request by tombujok**

### CODE REVIEW

**Comment by vbekiaris** [comment]
Looks good. Can you add some tests?

**Comment by tombujok** [comment]
Sure, they are on the EE side - otherwise it's impossible to test.

**Comment by tombujok** [comment]
run-lab-run

File: hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/SerializationServiceV1.java [hunk 1]
```diff
@@ -209,17 +213,26 @@ final PortableSerializer getPortableSerializer() {
     public ObjectDataInput initDataSerializableInputAndSkipTheHeader(Data data) throws IOException {
```
Comment (mmedenjak) [inline hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/SerializationServiceV1.java]: Update the Javadoc for the method

File: hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/DataSerializableSerializer.java [hunk 2]
```diff
@@ -46,8 +46,13 @@
  * <li>{@link IdentifiedDataSerializable}</li>
  * </ol>
  */
+@SuppressWarnings("checkstyle:npathcomplexity")
 final class DataSerializableSerializer implements StreamSerializer<DataSerializable>, TypedStreamDeserializer<DataSerializable> {
 
+    public static final byte IDS_FLAG = 1 << 0;
+    public static final byte EE_FLAG = 1 << 1;
+    public static final byte COMP_FLAG = 1 << 2;
```
Comment (mmedenjak) [inline hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/DataSerializableSerializer.java]: Maybe add short comment on what are these flags. I guess IDS is IdentifiedDataSerializable, EE is Enterprise but I couldn't figure out what COMP was.

File: hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/SerializationServiceV1.java (hunk 1 above)
Comment (tombujok) [inline hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/SerializationServiceV1.java]: Good catch. Thx

File: hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/DataSerializableSerializer.java (hunk 2 above)
Comment (tombujok) [inline hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/DataSerializableSerializer.java]: 👍

**Comment by tombujok** [comment]
run-lab-run

**Comment by tombujok** [comment]
run-lab-run

**Pull request merged**